pkg/operator/k8sutil/node.go

```diff
@@ -279,3 +280,74 @@ func NodeIsInRookNodeList(targetNodeName string, rookNodes []rookalpha.Node) boo
 	}
 	return false
 }
+
+// ValidateLabelKey validates Key of the Label against regex
+func ValidateLabelKey(s string) bool {
+	re := regexp.MustCompile(`^(((([a-zA-Z0-9]+[.]){1,}[a-zA-Z0-9]+){1}[\/]{1})?)([a-zA-Z0-9](([a-zA-Z0-9]*[\-_.][a-zA-Z0-9]+){1,})[a-zA-Z0-9]$)`)
```

Comment:
Can these regexs be replaced with the Kubernetes ones for label validation instead of having our "own" here?

Reply:
Can you point me to the source/example? I didn't find it.

Reply:
Should be in this file https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L88-L108

Though the variable naming seems a bit off. The validation pkg is probably already used somewhere in the Rook code, that may help you too.

Reply:
Should be in this file https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L88-L108

Though the variable naming seems a bit off. The validation pkg is probably already used somewhere in the Rook code, that may help you too.

Reply:
Thanks. Found and added native validation.